### PR TITLE
C-SPAN videos not working #18806 fix

### PR DESCRIPTION
**Pull request by Geremia**
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

- [X] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

- [X] Bug fix
---
fixes issue #18806 "C-SPAN videos not working" using [AlbinoDrought's patch](https://github.com/ytdl-org/youtube-dl/issues/18806#issuecomment-598996563)

### CODE REVIEW

File: youtube_dl/extractor/cspan.py [hunk 1]
```diff
@@ -145,6 +151,35 @@ def get_text_attr(d, attr):
         title = find_xpath_attr(doc, './/string', 'name', 'title').text
         thumbnail = find_xpath_attr(doc, './/string', 'name', 'poster').text
 
+        data = self._download_json(
+            'http://www.c-span.org/assets/player/ajax-player.php?os=android&html5=%s&id=%s' % (video_type, video_id),
```
Comment (vszakats) [inline youtube_dl/extractor/cspan.py:155]: This URL supports HTTPS, it'd be nice to use it.

Comment (tlsssl) [inline youtube_dl/extractor/cspan.py:155]: Can we do HTTPS here? (It is supported by the server)

**Comment by sbma44** [comment]
installed from author's branch, fixed the issue for me

**Comment by dstftw** [comment]
Superseded by 3e4e338133463e57125ade82fe5dfd57bb42dc1a. Thanks.

**Pull request closed without merging**